craft_application/commands/init.py

```diff
@@ -83,21 +84,18 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
             ),
         )
 
-    @property
-    def parent_template_dir(self) -> pathlib.Path:
+    def parent_template_dir(self) -> AbstractContextManager[pathlib.Path]:
```

Comment:
```suggestion
    @contextlib.contextmanager
    def parent_template_dir(self) -> pathlib.Path
        yield importlib.resources.path(self._app.name, "templates")
```
It's not letting me highlight the whole block for this suggestion (grr) so sorry if it looks weird. I would wrap this using a decorator instead, it's a more common pattern with our tools.

Reply:
Under this example wouldn't it need to be something like:

```python
with importlib.resources.path(...) as p:
    yield p
```

Otherwise you'd have a context manager that yields a context manager.

Reply:
*facepalm*, yeah, that's correct